src/main/java/net/minecraftforge/event/world/RaidEvent.java

```diff
@@ -0,0 +1,70 @@
+package net.minecraftforge.event.world;
+
+import net.minecraft.world.raid.Raid;
+import net.minecraftforge.eventbus.api.Cancelable;
+import net.minecraftforge.eventbus.api.Event;
+
+/**
+ *  RaidEvent along with its subevents gets fired whenever a {@link Raid} occurs.
+ */
+@Cancelable
+public class RaidEvent extends Event
+{
+   public final Raid raid;
+   
+   public RaidEvent(Raid raid)
+   {
+	  this.raid = raid;
+   }
+   
+   public Raid getRaid()
+   {
+	   return raid;
+   }
+   
+   /**
+    * This event is fired for every {@link Raid} tick.
+    * This event is {@link Cancelable}.
```

Comment:
What happens if you cancel it?

Reply:
skips ticks

Reply:
I know that. I mean you need to write that in the documentation. That's what it's there for.

Reply:
clarified it